server/utils/writeResponse.ts

```diff
@@ -0,0 +1,21 @@
+import { Response } from 'express';
+
+interface ResponseData {
+  httpStatus?: number;
+  messageKey?: string;
+  data: never;
+}
+
+function writeResponse(res: Response, responseData: ResponseData) {
+  const response = {
+    messageKey: responseData.messageKey,
+    data: responseData.data || undefined,
+  };
+
+  res
+    .status(responseData.httpStatus || 204)
+    .json(response)
+    .end();
+}
+
+export default writeResponse;
```

Comment:
Check version of this function in my branch. Seems little bit more friendly and typical.